src/Orleans/Runtime/GrainReferenceRuntime.cs

```diff
@@ -14,18 +15,18 @@ internal class GrainReferenceRuntime : IGrainReferenceRuntime
         private static ConcurrentDictionary<int, string> debugContexts = new ConcurrentDictionary<int, string>();
 
         private readonly Action<Message, TaskCompletionSource<object>> responseCallbackDelegate;
-        private readonly Logger logger;
+        private readonly ILogger logger;
         private readonly IInternalGrainFactory internalGrainFactory;
         private readonly SerializationManager serializationManager;
 
         public GrainReferenceRuntime(
-            Factory<string, Logger> loggerFactory,
+            ILoggerFactory loggerFactory,
```

Comment:
I believe this is being created by the container. If that's the case, you can just inject `ILogger<GrainReferenceRuntime>` (and assign it to the `ILogger` field) so that you skip the step of using the factory to create the actual logger.